### PR TITLE
destinationrule: support exportTo

### PR DESCRIPTION
**Pull request by yuval-k**
support destination rule export to field

### CODE REVIEW

**Review by stevenctl — APPROVED** [review]
LGTM!

File: projects/gateway2/proxy_syncer/destrule.go [hunk 1]
```diff
@@ -70,12 +74,29 @@ func NewEmptyDestRuleIndex() DestinationRuleIndex {
...
-			Hostname: d.Spec.GetHost(),
-		}}
+		exportTo := d.Spec.GetExportTo()
+		if len(exportTo) == 0 {
+			return []NsWithHostname{{
+				Ns:       exportAllNs,
+				Hostname: d.Spec.GetHost(),
+			}}
+		}
+		var keys []NsWithHostname
```
Comment (timflannagan) [inline projects/gateway2/proxy_syncer/destrule.go:88]: nit: can preallocate this array.

File: projects/gateway2/proxy_syncer/upstreams.go [hunk 2]
```diff
@@ -72,6 +72,9 @@ func NewPerClientEnvoyClusters(
 			hostname := ggv2utils.GetHostnameForUpstream(up.Inner)
 
 			destrule := destinationRulesIndex.FetchDestRulesFor(kctx, ucc.Namespace, hostname, ucc.Labels)
+			if destrule != nil {
+				logger.Debug("found destination rule", zap.Stringer("destrule", destrule), zap.String("targetHost", hostname))
```
Comment (timflannagan) [inline projects/gateway2/proxy_syncer/upstreams.go:76]: not blocking: I don't have my IDE in front of me. What type is destRule here (e.g. client.Object?). Is printing the entire object too verbose, and we're printing unnecessary info like metadata.managedFields?

**Review by timflannagan — COMMENTED** [review]
Didn't dive too deep - are those testdata file being used right now (I didn't see anything in ggv2setup_test.go but it's possible I'm blind), or are we planning on using them in a follow-up with the benefit to help unblock work streams?

**Review by timflannagan — APPROVED** [review]

**Comment by timflannagan** [comment]
(Whoops fat-fingered the comment instead of approved button.)

**Pull request merged**